### PR TITLE
feat: implement organisation deletion and e2e test

### PR DESCRIPTION
**Pull request by Iretoms**
​
## Description
I worked on the soft delete of organisation created by an authenticated user endpoint. This endpoint ensures the user is authenticated before the delete action is successful, it validates that the user created the organisation. It checks for invalid organisation Id and returns the appropriate errors. The deletion done by the user is temporary and can still be recovered.
​
## Related Issue (Link to Github issue)
 [#117](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/117)
​
## How Has This Been Tested?
The changes have been tested using the Go testing framework. The test environment was set up with Gin in test mode, mock data was created, and test cases for the DeleteOrganisation function were executed. These tests ensured the successful  deleted of organisation and set the deleted field to true in the test database indicating soft delete, also ensured to return the appropriate error messages for invalid organisation Id format, organisation not found and user not authorised to delete.
​
## Screenshots (if appropriate - Postman, etc):
Screenshot (36)
Screenshot (37)
Screenshot (38)
Screenshot (39)

​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: internal/models/organisation.go [hunk 1]
```diff
@@ -45,3 +46,13 @@ func (c *Organisation) CreateOrganisation(db *gorm.DB) error {
...
 	return nil
 }
+
+func (c *Organisation) Delete(db *gorm.DB) error {
+	err := db.Model(c).Update("deleted", true).Error
+
+	if err != nil {
+		return err
+	}
+
```
Comment (NwokoyeChigozie) [inline internal/models/organisation.go]: there should be a repository function for update

File: services/organisation/organisation.go [hunk 2]
```diff
@@ -72,3 +72,19 @@ func CreateOrganisation(req models.CreateOrgRequestModel, db *gorm.DB, userId st
...
 }
+
+func DeleteOrganisation(orgId string, userId string, db *gorm.DB) error {
+	var org models.Organisation
+	if err := db.Where("id = ? AND deleted = ?", orgId, false).First(&org).Error; err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return errors.New("organisation not found")
+		}
+		return err
+	}
```
Comment (NwokoyeChigozie) [inline services/organisation/organisation.go]: this should be in the model, also you should have repository function for those db calls

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]

File: services/organisation/organisation.go (hunk 2 above)
Comment (Iretoms) [reply services/organisation/organisation.go]: Alright, noted. On it

Comment (Iretoms) [reply services/organisation/organisation.go]: I've effected the change. Thank you

**Review by Cyberguru1 — DISMISSED** [review]
lgtm

**Comment by utuedey** [comment]
This branch has conflicts, kindly pull the new changes

File: pkg/repository/storage/postgresql/select.go [hunk 3]
```diff
@@ -215,6 +216,11 @@ func SelectFirstFromDb(db *gorm.DB, receiver interface{}) error {
 	return tx.Error
 }
 
+// func SelectOneFromDbAndUpdateField(db *gorm.DB, receiver interface{}, fieldName string, value interface{}, query interface{}, args ...interface{}) error {
```
Comment (Cyberguru1) [inline pkg/repository/storage/postgresql/select.go]: remove this if you no longer need it

Comment (Iretoms) [reply pkg/repository/storage/postgresql/select.go]: Alright, removed it

**Review by NwokoyeChigozie — APPROVED** [review]

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]
Fix Test

**Comment by Iretoms** [comment]
> Fix Test

done

**Pull request closed without merging**